roadmap/implementors-guide/guide.md

```diff
@@ -1003,7 +1026,365 @@ Dispatch a `PovFetchSubsystemMessage(relay_parent, candidate_hash, sender)` and
+  - In order to receive a `Seconded` message we have the on corresponding chain head in our view
+  - In order to receive an `Invalid` or `Valid` message we must have received the corresponding `Seconded` message.
+
+And respect this data-dependency order from our peers by respecting their views. This subsystem is responsible for checking message signatures.
+
+No jobs, `StartWork` and `StopWork` pulses are used to control neighbor packets and what we are currently accepting.
+
+----
+
+### PoV Distribution
+
+#### Description
+
+This subsystem is responsible for distributing PoV blocks. For now, unified with statement distribution system.
+
+#### Protocol
+
+Handle requests for PoV block by candidate hash and relay-parent.
+
+#### Functionality
+
+Implemented as a gossip system, where `PoV`s are not accepted unless we know a `Seconded` message.
+
+[TODO: this requires a lot of cross-contamination with statement distribution even if we don't implement this as a gossip system. In a point-to-point implementation, we still have to know _who to ask_, which means tracking who's submitted `Seconded`, `Valid`, or `Invalid` statements - by validator and by peer. One approach is to have the Statement gossip system to just send us this information and then we can separate the systems from the beginning instead of combining them]
+
+----
+
+### Availability Distribution
+
+#### Description
+
+Distribute availability erasure-coded chunks to validators.
+
+After a candidate is backed, the availability of the PoV block must be confirmed by 2/3+ of all validators. Validating a candidate successfully and contributing it to being backable leads to the PoV and erasure-coding being stored in the availability store.
+
+#### Protocol
+
+`ProtocolId`:`b"avad"`
+
+Input:
+  - NetworkBridgeUpdate(update)
+
+Output:
+  - NetworkBridge::RegisterEventProducer(`ProtocolId`)
+  - NetworkBridge::SendMessage(`[PeerId]`, `ProtocolId`, `Bytes`)
+  - NetworkBridge::ReportPeer(PeerId, cost_or_benefit)
+  - AvailabilityStore::QueryPoV(candidate_hash, response_channel)
+  - AvailabilityStore::StoreChunk(candidate_hash, chunk_index, inclusion_proof, chunk_data)
+
+#### Functionality
+
+Register on startup an event producer with  `NetworkBridge::RegisterEventProducer`.
+
+For each relay-parent in our local view update, look at all backed candidates pending availability. Distribute via gossip all erasure chunks for all candidates that we have to peers.
+
+We define an operation `live_candidates(relay_heads) -> Set<AbridgedCandidateReceipt>` which returns a set of candidates a given set of relay chain heads implies should be currently gossiped. This is defined as all candidates pending availability in any of those relay-chain heads or any of their last `K` ancestors. We assume that state is not pruned within `K` blocks of the chain-head.
```

Comment:
> which returns a set of candidates a given set of relay chain heads implies should be currently gossiped.

This sentence is confusing to me. Are `implies` and `should` at the right spot?

Reply:
No, they are not :)

Reply:
```suggestion
We define an operation `live_candidates(relay_heads) -> Set<AbridgedCandidateReceipt>` which returns a set of candidates a given set of relay chain heads that implies a set of candidates whose availability chunks should be currently gossiped. This is defined as all candidates pending availability in any of those relay-chain heads or any of their last `K` ancestors. We assume that state is not pruned within `K` blocks of the chain-head.
```